app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SessionsController < ApplicationController
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      flash[:success] = "Welcome, #{user.email}!"
+      redirect_to(dashboard_path)
+    else
+      invalid_credentials_redirect
+    end
+  end
+
+  private
+
+  def invalid_credentials_redirect
+    if !User.find_by(email: params[:email])
+      flash[:error] = 'No user exists with that email address.'
+    else
+      flash[:error] = 'Incorrect password'
+    end
+    redirect_to(root_path)
+  end
```

Comment:
Moved this into a private method as rubocop said living in 1 method violated method length.

Reply:
Also, could we talk through this one tomorrow? 
When I view params, the raw password is visible. That's like bad, right? (in my paris hilton voice) 
In past projects, my partners and I have never addressed this but I'm curious if you know about it.